File: Algorytmy/NWD&NWW.cpp
// Algorytm iteracyjny obliczania największego wspólnego dzielnika (NWD)
int NWD_iter(int a, int b){
    while(a != b){
        if(a > b)
            a -= b; // Jeśli a > b, odejmujemy b od a
        else
            b -= a; // W przeciwnym razie odejmujemy a od b
    }
    return a; // Zwracamy a (które jest NWD)
}

// Algorytm rekurencyjny obliczania największego wspólnego dzielnika (NWD)
int NWD_rek(int a, int b){
    if(b == 0)
        return a; // Jeśli b jest zerem, zwracamy a (ostatnie niezerowe a)
    else
        return NWD_rek(b, a % b); // W przeciwnym razie wywołujemy rekurencyjnie dla (b, a % b)
}

// Funkcja obliczająca najmniejszą wspólną wielokrotność (NWW)
int NWW(int a, int b){
    return a * b / NWD_iter(a, b); // Obliczamy NWW za pomocą iteracyjnie lub rekurencyjnie obliczonego NWD
}

File: Algorytmy/NWD&NWW.py
# Iteracyjny algorytm obliczania największego wspólnego dzielnika (NWD)
def NWD_iter(a, b):
    while b != 0:
        a, b = b, a % b  # Zastępuje a przez b, a b przez resztę z dzielenia a przez b
    return a  # Zwraca a (które jest NWD)

# Rekurencyjny algorytm obliczania największego wspólnego dzielnika (NWD)
def NWD_rek(a, b):
    if b == 0:
        return a  # Jeśli b jest zerem, zwraca a (ostatnie niezerowe a)
    else:
        return NWD_rek(b, a % b)  # W przeciwnym razie wywołuje rekurencyjnie dla (b, a % b)

# Funkcja obliczająca najmniejszą wspólną wielokrotność (NWW)
def NWW(a, b):
    return a * b / NWD_iter(a, b)  # Oblicza NWW za pomocą iteracyjnie lub rekurencyjnie obliczonego NWD